src/app/components/wrapper-components/tutorial-game-wrapper/tutorial-game-wrapper.component.ts

```diff
@@ -97,19 +97,24 @@ export class TutorialGameWrapperComponent extends GameWrapper<TutorialPlayer> im
         const currentStep: TutorialStep = this.steps[this.stepIndex];
         this.currentMessage = currentStep.instruction;
         this.currentReason = MGPOptional.empty();
-        let motherOpt: MGPOptional<MGPNode<Rules<Move, GameState>, Move, GameState>>;
-        if (currentStep.previousState.isPresent()) {
+        this.createCurrentStepNode(currentStep);
+        this.gameComponent.updateBoard();
+        this.cdr.detectChanges();
+    }
+    private createCurrentStepNode(currentStep: TutorialStep): void {
+        let motherOpt: MGPOptional<MGPNode<Rules<Move, GameState>, Move, GameState>> = MGPOptional.empty();
+        let previousMove: MGPOptional<Move> = MGPOptional.empty();
+        for (let i: number = 0; i < currentStep.previousState.length; i++) {
+            const state: GameState = currentStep.previousState[i];
+            previousMove = MGPOptional.ofNullable(currentStep.previousMove[i - 1]);
```

Comment:
hmm donc quand i = 0 on accède à previousMove[-1] ? c'pas comme en python j'imagine donc ça donne un undefined, non ?

Reply:
ja.

Reply:
pourquoi pas simplement commencer à `i = 1` du coup ?

Reply:
out of date